google/resumable_media/_download.py

```diff
@@ -162,10 +185,15 @@ def _process_response(self, response):
         self._finished = True
         _helpers.require_status_code(
             response, _ACCEPTABLE_STATUS_CODES, self._get_status_code)
+        if self._stream is not None:
+            self._write_to_stream(response)
```

Comment:
It's fine for this to indirectly perform network IO, just as long as this class provides some means of replacing the network IO in subclasses.